pkg/agent/watch.go

```diff
@@ -300,24 +300,42 @@ func makeVMRestartMetrics(vm *vmv1.VirtualMachine) []vmMetric {
 	}
 }
 
+type gaugeSpec struct {
```

Comment:
suggestion: Could you add some comments explaining what `gaugeSpec` does / is for?

I'm having a little trouble wrapping my head around it :sweat_smile:

Reply:
Expanded comments around `gaugeSpec` and `getGaugeSpecs`